src/storage/leveldb/store.ts

```diff
@@ -156,10 +151,12 @@ export default class LevelDBStore implements IStore {
         }
       }
     }
-    if ((await this.walletIndex.getCurrentAddressIndex()) < maxIndex) {
-      await this.walletIndex.setCurrentAddressIndex(Math.min(maxIndex + 1, await this.walletIndex.getLastLoadedAddressIndex()));
+    if (maxIndex >= 0) {
```

Comment:
I feel it would be nice to have a comment about why this `maxIndex >= 0` validation is necessary

If I understand correctly, we're just trying to avoid calculating addresses when the wallet has never used any address beyond index 0.

Reply:
Well, we use -1 as a starting point and if no address is used (should not happen, but it did on some tests) we will try to save -1 on the leveldb, but we encode it as unsigned int, which throws an error since -1 is negative.

This check is only to avoid this error, since negative address index should be impossible in bip39 we are not avoiding any actual cases.